### PR TITLE
[AUD-1598] Add NotificationSettingsScreen

### PR DESCRIPTION
**Pull request by dylanjeffers**
### Description

- Adds notification settings screen
- Moves settings state to common
- Adds switch core component

https://user-images.githubusercontent.com/8230000/155601269-89bbab58-922e-4efa-aad1-6ba4f797c05d.mov

### CODE REVIEW

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/dj-aud-1598-notification-screen

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/dj-aud-1598-notification-screen

File: packages/web/src/pages/settings-page/components/desktop/SettingsPage.tsx [hunk 1]
```diff
@@ -36,7 +36,7 @@ import {
   Notifications,
   BrowserNotificationSetting,
   EmailFrequency
-} from '../../store/types'
+} from '../../../../common/store/pages/settings/types'
```
Comment (sliptype) [inline packages/web/src/pages/settings-page/components/desktop/SettingsPage.tsx:39]:
```suggestion
} from 'common/store/pages/settings/types'
```

File: packages/web/src/pages/settings-page/components/mobile/SettingsPage.tsx [hunk 2]
```diff
@@ -32,7 +32,7 @@ import {
   PushNotificationSetting,
   PushNotifications,
   Cast
-} from '../../store/types'
+} from '../../../../common/store/pages/settings/types'
```
Comment (sliptype) [inline packages/web/src/pages/settings-page/components/mobile/SettingsPage.tsx:35]:
```suggestion
} from 'common/store/pages/settings/types'
```

**Review by sliptype — APPROVED** [review]
Calm down dude, flying through these screens!

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/dj-aud-1598-notification-screen

File: packages/mobile/src/components/core/Switch.tsx [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  const { neutralLight6, neutralLight9, secondary } = useThemeColors()
+  const [isEnabledState, setIsEnabled] = useToggle(defaultValue)
+
+  const isEnabled = value ?? isEnabledState
+
+  const handleValueChange = useCallback(
+    (value: boolean) => {
+      onValueChangeProp?.(value)
+      setIsEnabled(value)
+      light()
```
Comment (raymondjacobson) [inline packages/mobile/src/components/core/Switch.tsx:29]: amen. good lookin here. we should add more haptics generally

File: packages/mobile/src/screens/settings-screen/EmailFrequencyControlRow.tsx [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+import { SegmentedControl } from 'app/components/core'
+import { useDispatchWeb } from 'app/hooks/useDispatchWeb'
+import { useSelectorWeb } from 'app/hooks/useSelectorWeb'
+
+import { SettingsRowLabel } from './SettingRowLabel'
+import { SettingsRow } from './SettingsRow'
+import { SettingsRowContent } from './SettingsRowContent'
+
+const messages = {
+  emailFrequency: '‘What You Missed’ Email Frequency',
```
Comment (raymondjacobson) [inline packages/mobile/src/screens/settings-screen/EmailFrequencyControlRow.tsx:16]: Should this be back tick on the outside and regular single quote on the inside?

**Review by raymondjacobson — APPROVED** [review]

File: packages/mobile/src/components/core/Switch.tsx [hunk 5]
```diff
@@ -0,0 +1,44 @@
+import { useCallback } from 'react'
+
+import { Switch as RNSwitch, SwitchProps as RNSwitchProps } from 'react-native'
```
[the remark this answers is not in the document]
Comment (raymondjacobson) [reply packages/mobile/src/components/core/Switch.tsx:3]: A note on switch -- I need to play around with this a bit, but we were very deliberate on the animations the first time around when we built these. Active state slightly expands the inner dot. Just want to flag to confirm behavior. Good to ship this either way, but maybe will need to iterate

File: packages/mobile/src/components/core/Switch.tsx (hunk 3 above)
Comment (dylanjeffers) [reply packages/mobile/src/components/core/Switch.tsx:29]: totally agree! esp on the favorite button

File: packages/mobile/src/screens/settings-screen/EmailFrequencyControlRow.tsx (hunk 4 above)
Comment (dylanjeffers) [reply packages/mobile/src/screens/settings-screen/EmailFrequencyControlRow.tsx:16]: yeah exactly, good call

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/dj-aud-1598-notification-screen

**Pull request merged**